lib/neuron/fragment.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Neuron.Fragment do
+  alias Neuron.Store
+
+  @doc ~S"""
+  registers a fragment that will automatically be added to future mutations and queries that require it
+
+  ## Example
+
+  ```elixir
+  Neuron.register("\"\"
+    NameParts on Person {
+      firstName
+      lastName
+    }
+  \"\"")
+  ```
+  """
+
+  @spec register(query_string :: String.t()) :: :ok
+  def register(query_string), do: register(:global, query_string)
+
+  @spec register(context :: :global | :process, query_string :: String.t()) :: :ok
+  def register(context, query_string) do
+    fragment = query_string |> process_fragment_string
+    stored_fragments = Store.get(context, :fragments, [])
+    Store.set(context, :fragments, [fragment | stored_fragments])
+  end
+
+  defp process_fragment_string(query_string) do
+    fragment_name = Regex.run(~r/^(?:\W*)(\w+)/, query_string) |> List.last() |> String.to_atom()
+
+    fragment = query_string |> construct_fragment_string
+
+    {fragment_name, fragment}
+  end
+
+  def insert_into_query(query_string) do
```

Comment:
Could we add `@doc false` here? Even though this function needs to be public, it's not intended to be used outside of the library.